static/src/javascripts/lib/add-event-listener.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+
+let supportsOptions = false;
+
+try {
+    const opts = Object.defineProperty(
+        {},
+        'passive',
+        ({
+            get() {
+                supportsOptions = true;
+            },
+        }: Object) // https://github.com/facebook/flow/issues/285
+    );
+    window.addEventListener('test', null, opts);
+} catch (e) {
+    /* noop */
+}
+
+const addEventListener = (
+    node: Node,
+    name: string,
+    handler: (e: Event) => void,
+    options: Object = {}
+): void => {
+    if (node) {
```

Comment:
I thought that was the whole purpose of Flow, that signatures allow it to run a summative flow analysis and enforce constraints such as mandatory arguments.